sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1272,8 +1272,11 @@ object InferFiltersFromConstraints extends Rule[LogicalPlan]
   private def inferFilters(plan: LogicalPlan): LogicalPlan = plan.transformWithPruning(
     _.containsAnyPattern(FILTER, JOIN)) {
     case filter @ Filter(condition, child) =>
-      val newFilters = filter.constraints --
```

Comment:
is it possible to override `--` in `ExpressionSet`?

Reply:
cc @LuciferYang

Reply:
Yes, after SPARK-39520, this issue no longer exists， and @HyukjinKwon already mark SPARK-39523 as RESOLVED

Reply:
It seems https://github.com/apache/spark/pull/36925 fixed this issue ? @LuciferYang

Reply:
[SPARK-39520](https://issues.apache.org/jira/browse/SPARK-39520) was not merged to 3.3, do we have this issue in 3.3?

Reply:
> It seems #36925 fixed this issue ? @LuciferYang

Yes

Reply:
> [SPARK-39520](https://issues.apache.org/jira/browse/SPARK-39520) was not merged to 3.3, do we have this issue in 3.3?

Let me double check this

Reply:
@cloud-fan @LuciferYang I have tested and the issue no longer exists. I will close this PR.

Reply:
> [SPARK-39520](https://issues.apache.org/jira/browse/SPARK-39520) was not merged to 3.3, do we have this issue in 3.3?

@cloud-fan 3.3 no this issue due to SPARK-38836 not in branch-3.3, the `--` method hasn't been deleted, and I manually check the relevant suites(inlcude TPCDSV1_4_PlanStabilitySuite, AdaptiveQueryExecSuite, ExpressionSetSuite, JDBCV2Suite, GeneratorFunctionSuite), all passed.